### DISCUSSION TITLE
Branch picker: Order is wrong; exact matches should be listed first

### DISCUSSION THREAD

**Discussion by nh2** [community/community#4594] [Pull Requests]
I think something is off with the order of branches shown in the branch picker (both in code browsing, and on the pull request page):

* It should sort by lexicographical order. Consequently, exact matches should be listed first.

But they are not. It puts the exact match at the very end of the list:

```
staging-next
staging-patchelf
...
staging
```

Only the last entry (the exact match), is out of lexicographical order.

The order should be lexicographical. That would also be better UX, so that one doesn't have to scroll all the way down to find the exact match one entered.

Screenshot of the problem:

![image](https://user-images.githubusercontent.com/399535/125179607-0bb8e780-e1f0-11eb-98d6-9eeb552b0ec1.png)

Demonstration that lexicographical order puts exact matches first (in `python`):

```python
>>> sorted(["mybranch-123", "mybranch"])
['mybranch', 'mybranch-123']
```

Thanks!

**Comment by micaelbergeron** [comment]
@nh2 I agree with you :100:

I think we should also be able to use `Enter` to select the current highlighted choice.

**Comment by micaelbergeron** [comment]
As of today, it seems like the branch picker behaves this way :D

**Comment by nh2** [comment]
I observe a change now, but **only in the code browsing**, not in the pull request branch picker:

Code browsing (improved):

![image](https://user-images.githubusercontent.com/399535/136234322-0bf339d8-9203-4dcf-97c5-ec5ef348623a.png)


Pull request (unchanged):

![image](https://user-images.githubusercontent.com/399535/136234228-7a417ef8-80ba-4def-8739-88cf8f0562c7.png)
